google/cloud/forseti/services/notifier/service.py

```diff
@@ -81,25 +81,39 @@ def Run(self, request, _):
         """
         progress_queue = Queue()
 
-        LOGGER.info('Run notifier service with inventory index id: %s',
-                    request.inventory_index_id)
+        if request.scanner_index_id and request.inventory_index_id:
```

Comment:
Just having the logger.exception message is not enough.  The notifier will still run at the bottom.  You should just return a response right away, to stop the execution here.